2018/solution3.py

```diff
@@ -0,0 +1,63 @@
+import sys
+
+from common import read_strings
+
+
+def count_overlap(claims):
+    filled = populate_map(parse_input(claims))
+    return len([x for x in filled if filled[x] > 1])
+
+def populate_map(claims):
+    filled = {}
+    for claim_id in claims:
+        claim = claims[claim_id]
+        for x in range(claim[0][0], claim[0][0] + claim[1][0]):
+            for y in range(claim[0][1], claim[0][1] + claim[1][1]):
+                if (x, y) not in filled:
+                    filled[(x, y)] = 0
+                filled[(x, y)] += 1
+    return filled
+
+def get_0_overlap_id(claims):
+    parsed_data = parse_input(claims)
+    filled = populate_map(parsed_data)
+    return next(claim_id
+                for claim_id in parsed_data
+                if not has_overlap(filled, parsed_data[claim_id]))
+
+def has_overlap(filled, claim):
+    for x in range(claim[0][0], claim[0][0] + claim[1][0]):
+        for y in range(claim[0][1], claim[0][1] + claim[1][1]):
+            if filled[(x, y)] > 1:
+                return True
+    return False
+
+def parse_input(claims):
```

Comment:
I had a tough time following all the parsing logic.  I think there were a lot of moving parts to it